### DISCUSSION TITLE
wallet provider

### DISCUSSION THREAD

**Discussion by dev-aly3n** [wevm/wagmi#2237] [Q&A]
how to use user's wallet as a provider without using alchemy, infura or public providers?

I remember something similar when I used wagmi in the last year. but maybe it changed with updates.

**for now passing provider to createClient is mandatory.

**Comment by jxom** [comment]
There is no first-class API to do this in wagmi, and it is strongly discouraged as some wallets do not provide (or restrict) certain methods like `eth_call`, `eth_getLogs`, `eth_newFilter`, etc which are essential for contract interactions, block/contract events, etc.
